src/kernel/TurbKineticEnergyKsgsDesignOrderSrcElemKernel.C

```diff
@@ -130,8 +130,12 @@ TurbKineticEnergyKsgsDesignOrderSrcElemKernel<AlgTraits>::execute(
         Pk += w_dudx[i][j]*(w_dudx[i][j] + w_dudx[j][i]);
       }
     }
-    Pk *= tviscIp;
-    
+    Pk *= stk::math::max(tviscIp,0);
+    tkeIp = stk::math::max(tkeIp,0);
+    dualNodalVolIp = stk::math::max(dualNodalVolIp,0);
+    rhoIp = stk::math::max(rhoIp,0);
```

Comment:
Can we clip this to something more reasonable? I think that, at present, we only have the ability to clip Z to a reference value. However, perhaps this is not required. 

I guess that I am surprised that rho and dualNodalVolIp need clipping - especially since they should not hover close to 0.0. Have you seen this in tests or is this just a worst case approach?